portal/tests/test_independent_student_new.py

```diff
@@ -116,6 +121,64 @@ def test_reset_password_fail(self):
 
         assert len(mail.outbox) == 0
 
+    def test_join_class_accepted(self):
+        teacher_email, teacher_password = signup_teacher_directly()
+        create_organisation_directly(teacher_email)
+        klass, class_name, accesss_code = create_class_directly(teacher_email)
+        create_school_student_directly(accesss_code)
+        klass.always_accept_requests = True
+        klass.save()
+
+        homepage = self.go_to_homepage()
+
+        play_page, student_name, student_username, student_email, password = create_independent_student(homepage)
+
+        page = play_page \
+            .independent_student_login(student_username, password) \
+            .go_to_join_a_school_or_club_page() \
+            .join_a_school_or_club(accesss_code)
+
+        page.logout()
+
+        page = self.go_to_homepage()
+
+        class_page = page \
```

Comment:
maybe, rename class_page by something else (this is also something else imported elsewhere so might conflict?)